ros2_socketcan/src/socket_can_sender.cpp

```diff
@@ -138,8 +147,30 @@ void SocketCanSender::send_impl(
   data_frame.can_dlc = static_cast<decltype(data_frame.can_dlc)>(length);
   //lint -e{586} NOLINT data_frame is a stack variable; guaranteed not to overlap
   (void)std::memcpy(static_cast<void *>(&data_frame.data[0U]), data, length);
-  const auto bytes_sent = ::send(m_file_descriptor, &data_frame, sizeof(data_frame), flags);
-  if (0 > bytes_sent) {
+
+  int number = 0;
+  ssize_t nbytes = 0;
+
+  struct can_frame * p = &data_frame;
+  ulong bytes_sent = 0;
+  while (bytes_sent < sizeof(data_frame) && -1 != nbytes) {
+    for (int i = 0; i < ERR_AGAIN_RETRY_CNT; i++) {
+      nbytes = ::send(
+        m_file_descriptor, (char *)p + bytes_sent, sizeof(data_frame) - bytes_sent,
+        flags);
```

Comment:
```suggestion
      nbytes = ::send(
        m_file_descriptor, reinterpret_cast<char *>(p)  + bytes_sent, sizeof(data_frame) - bytes_sent,
        flags);
```

Related test error: https://github.com/autowarefoundation/ros2_socketcan/actions/runs/9974311030/job/27561524880?pr=31#step:5:1406